mmv1/third_party/terraform/fwprovider/framework_validators.go

```diff
@@ -84,3 +84,34 @@ func (v nonnegativedurationValidator) ValidateString(ctx context.Context, reques
 func NonNegativeDurationValidator() validator.String {
 	return nonnegativedurationValidator{}
 }
+
+// Non Empty String Validator
+type nonEmptyStringValidator struct {
+}
+
+// Description describes the validation in plain text formatting.
+func (v nonEmptyStringValidator) Description(_ context.Context) string {
+	return "value expected to be a string that isn't an empty string"
+}
+
+// MarkdownDescription describes the validation in Markdown formatting.
+func (v nonEmptyStringValidator) MarkdownDescription(ctx context.Context) string {
+	return v.Description(ctx)
+}
+
+// ValidateString performs the validation.
+func (v nonEmptyStringValidator) ValidateString(ctx context.Context, request validator.StringRequest, response *validator.StringResponse) {
+	if request.ConfigValue.IsNull() || request.ConfigValue.IsUnknown() {
+		return
+	}
+
+	value := request.ConfigValue.ValueString()
+
+	if value == "" {
+		response.Diagnostics.AddError("please provide a value that isn't an empty string to this field", fmt.Sprintf("%s was set to `%s`", request.Path, value))
```

Comment:
I'm happy to get feedback on the phrasing of this validation error message - I just put in something that worked!

Reply:
I don't have strong preference. Maybe we can make the error message match with the others, such as https://github.com/GoogleCloudPlatform/magic-modules/blob/bcb9cad3603c2843b3ffc66ff20604a3cdca96da/mmv1/third_party/terraform/fwprovider/framework_validators.go#L75

Reply:
That's a good idea - `expected a non-empty string` is shorter and matches the existing style.